src/components/TodoItem.tsx

```diff
@@ -0,0 +1,69 @@
+import React, { useState } from 'react';
+import cn from 'classnames';
+import { Todo, Error } from '../types/Todo';
+import * as todosService from '../api/todos';
+
+type Props = {
+  todos?: Todo[],
+  setTodos?: (value: Todo[]) => void,
+  todo: Todo,
+  isNewTodoLoading?: boolean
+  setHasError: (value: Error) => void,
```

Comment:
Try not to pass all the todos and setter to the todoitem, also move request to the component. this will reduce the number of props you pass to
```suggestion
  todo: Todo,
  isNewTodoLoading?: boolean
```